### PR TITLE
Implemented MVP and network setup

### PR DESCRIPTION
**Pull request by denpronin**
Forgot to rename NetworkService

### CODE REVIEW

File: app/src/main/java/com/pronin/weather/network/NetworkService.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    public CurrentWeatherApi getApi() {
+        return retrofit.create(CurrentWeatherApi.class);
+    }
+
+    public static void getCurrentWeather(String apikey, GetCurrentWeatherListener callback) {
+        Thread thread = new Thread(() -> {
+            Map<String, String> options = new HashMap<>();
+            options.put("appid", apikey);
+            options.put("lat", "55");
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/weather/network/NetworkService.java:40]: remove magic nambers

File: app/src/main/java/com/pronin/weather/network/NetworkService.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            options.put("units", "metric");
+            options.put("exclude", "minutely,alerts");
+            options.put("lang", "ru");
+            Call<CurrentWeather> currentWeather = getInstance().getApi()
+                    .getWeather(options);
+            try {
+                Response<CurrentWeather> currentWeatherResponse = currentWeather.execute();
+                callback.onGotCurrentWeather(currentWeatherResponse.body());
+            } catch (IOException ex) {
+                ex.printStackTrace();
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/weather/network/NetworkService.java:51]: add invoke callback function onError() and show some toast

File: app/src/main/java/com/pronin/weather/network/NetworkService.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                .baseUrl(BASE_URL)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+    }
+
+    public CurrentWeatherApi getApi() {
+        return retrofit.create(CurrentWeatherApi.class);
+    }
+
+    public static void getCurrentWeather(String apikey, GetCurrentWeatherListener callback) {
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/weather/network/NetworkService.java:36]: change this callback flow to rxJava

File: app/src/main/java/com/pronin/weather/network/NetworkService.java (hunk 2 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/weather/network/NetworkService.java:51]: added it and checking of internet connection

File: app/src/main/java/com/pronin/weather/network/NetworkService.java (hunk 1 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/weather/network/NetworkService.java:40]: added location

**Pull request merged**